pkg/driver/controller.go

```diff
@@ -169,12 +224,18 @@ func CreateZFSVolume(req *csi.CreateVolumeRequest) (string, error) {
 			"not able to provision the volume %s", err.Error())
 	}
 
+	if _, ok := parameters["wait"]; ok {
+		if err := waitForReadyVolume(volName); err != nil {
+			return "", err
+		}
+	}
+
 	return selected, nil
 }
 
 // CreateVolClone creates the clone from a volume
 func CreateVolClone(req *csi.CreateVolumeRequest, srcVol string) (string, error) {
-	volName := req.GetName()
+	volName := strings.ToLower(req.GetName())
```

Comment:
With this change are there any backward compatiblity check that need to be verified. Like, creating a snapshot or clone in older release and after upgrading trying to create new snapshot or delete old snap or clone. 

If the ids were always - pvc-uuid.. may be it is not an issue. just checking here.

Reply:
Yes, it was always pvc-uuid.